src/gui/vector/qgsattributesformproperties.cpp

```diff
@@ -1988,3 +2010,283 @@ void QgsAttributesFormProperties::updatedFields()
+      return;
+
+    const QTreeWidgetItem *itemLayout = mFormLayoutTree->selectedItems().at( 0 );
+    const DnDTreeItemData itemDataLayout = itemLayout->data( 0, DnDTreeRole ).value<DnDTreeItemData>();
+
+    QDomElement displayElement = doc.createElement( QStringLiteral( "widgetDisplay" ) );
+    displayElement.setAttribute( QStringLiteral( "showLabel" ), itemDataLayout.showLabel() );
+    displayElement.setAttribute( QStringLiteral( "horizontalStretch" ), itemDataLayout.horizontalStretch() );
+    displayElement.setAttribute( QStringLiteral( "verticalStretch" ), itemDataLayout.verticalStretch() );
+    displayElement.appendChild( itemDataLayout.labelStyle().writeXml( doc ) );
+    documentElement.appendChild( displayElement );
+  }
+
+  doc.appendChild( documentElement );
+
+  QMimeData *mimeData = new QMimeData;
+  mimeData->setData( QStringLiteral( "application/x-qgsattributetabledesignerelementclipboard" ), doc.toByteArray() );
+  QClipboard *clipboard = QApplication::clipboard();
+  clipboard->setMimeData( mimeData );
+}
+
+void QgsAttributesFormProperties::pasteWidgetConfiguration()
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  if ( mAvailableWidgetsTree->selectedItems().count() != 1 )
+    return;
+
+  QTreeWidgetItem *item = mAvailableWidgetsTree->selectedItems().at( 0 );
+
+  const QString fieldName = item->data( 0, FieldNameRole ).toString();
+  const int fieldIndex = mLayer->fields().indexOf( fieldName );
+
+  if ( fieldIndex < 0 )
+    return;
+
+  // Get base config from target item and ovewrite settings when possible
+  FieldConfig config = item->data( 0, FieldConfigRole ).value<FieldConfig>();
+
+  QDomDocument doc;
+  QClipboard *clipboard = QApplication::clipboard();
+  if ( doc.setContent( clipboard->mimeData()->data( QStringLiteral( "application/x-qgsattributetabledesignerelementclipboard" ) ) ) )
+  {
+    QDomElement docElem = doc.documentElement();
+    if ( docElem.tagName() != QLatin1String( "FormWidgetClipboard" ) )
+      return;
+
+    // We avoid pasting if origin and target fields are the same
+    const QString sourceFieldName = docElem.attribute( QStringLiteral( "name" ) );
+    if ( sourceFieldName == fieldName )
+      return;
+
+    // When pasting, the target item has already been selected and
+    // has triggered attribute type dialog loading. Therefore, we'll
+    // only overwrite GUI settings instead of destroying and recreating
+    // the whole dialog.
+
+    // Editor widget configuration
+    const QDomElement fieldWidgetElement = docElem.firstChildElement( QStringLiteral( "editWidget" ) );
+    if ( !fieldWidgetElement.isNull() )
+    {
+      const QString widgetType = fieldWidgetElement.attribute( QStringLiteral( "type" ) );
+
+      // Only paste if source editor widget type is supported by target field
+      const QgsEditorWidgetFactory *factory = QgsGui::editorWidgetRegistry()->factory( widgetType );
+      if ( factory->supportsField( mLayer, fieldIndex ) )
```

Comment:
could we get a form of user feedback on the reason it fails?

Reply:
Sure, I'd thought of just logging it so that it's shown in the QGIS log window.

What would be your preferred way?

Reply:
There is the message bar available.

But I thought there would be a message bar in the dialog, that might be worth adding one like in QgsAttributeForm.